spec/services/sign_in/token_response_generator_spec.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe SignIn::TokenResponseGenerator do
+  subject(:generator) { described_class.new(valid_params:, cookies:) }
+
+  let(:cookies) { double('cookies') }
+  let(:valid_params) { {} }
+
+  before { allow(Rails.logger).to receive(:info) }
+
+  describe '#perform' do
+    context 'when grant_type is AUTH_CODE_GRANT' do
+      let(:grant_type) { SignIn::Constants::Auth::AUTH_CODE_GRANT }
+      let(:valid_params) do
+        {
+          grant_type:,
+          code: 'some-code',
+          code_verifier: 'some-code-verifier',
+          client_assertion: 'some-client-assertion',
+          client_assertion_type: 'some-client-assertion-type',
+          client_id: 'some-client-id'
+        }
+      end
+      let(:access_token) { 'some-access-token' }
+      let(:validated_credential) { double('validated_credential') }
```

Comment:
Any reason not to just use the factories for all this? I feel like this is forcing a ton of mocking and I feel like it's quite brittle

Reply:
yeah good call. I guess i didn't realize there were factories for these